ThumbnailItem/ThumbnailItem.js

```diff
@@ -13,25 +17,80 @@ import kind from '@enact/core/kind';
 import PropTypes from 'prop-types';
 import React from 'react';
 
-import componentCss from './ThumbnailItem.module.less';
 import Item from '../Item';
 
+import componentCss from './ThumbnailItem.module.less';
 
 /**
  * A stateless, unfocusable item that can display a thumbnail.
  *
  * @class ThumbnailItemBase
  * @memberof agate/ThumbnailItem
  * @extends agate/Item.Item
- * @agate
+ * @ui
  * @public
  */
 const ThumbnailItemBase = kind({
 	name: 'ThumbnailItem',
 
 	propTypes: /** @lends agate/ThumbnailItem.ThumbnailItemBase.prototype */ {
+		/**
+		 * The main content displayed with the thumbnail.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		content: PropTypes.string,
+
+		/**
+		 * Customizes the component by mapping the supplied collection of CSS class names to the
+		 * corresponding internal elements and states of this component.
+		 *
+		 * The following classes are supported:
+		 *
+		 * * `thumbnailItem` - The root component class
+		 * * `content` - The content component class
+		 * * `subContent` - The subContent component class
+		 * * `roundThumbnail` - Applied when `roundThumbnail="true"`
+		 * * `thumbnail` - The thumbnail component class
+		 *
+		 * @type {Object}
+		 * @public
+		 */
 		css: PropTypes.object,
-		src: PropTypes.string
+
+		/**
+		 * Applies a round thumbnail to the ThumbnailItem.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		roundThumbnail: PropTypes.bool,
```

Comment:
I think that it might be better to have the `type` prop instead of `roundThumbnail`. So the `type` could have a `round` and a `square`. By doing it, we could have more types. For boolean type, we could have only two types.

Reference:
https://github.com/enactjs/agate/blob/develop/Button/Button.js#L187
```
type: PropTypes.oneOf(['grid', 'standard'])
```

Reply:
Oh, there is a better way.

I hope that we could support `imageComponent` prop.
The round and square images are very popular in silicon skin. So let's make that image as a component and make it as the `imageComponent` prop default value of the ThumbnailItem.

The default value of the `imageComponent` for other skins would be
```
<img
	className={css.thumbnail}
	src={src}
/>
```

Please let me know your thought.

Reply:
I applied the first approach